src/Uno.UI.Runtime.Skia.Wpf/Islands/UnoXamlHostBase.Focus.cs

```diff
@@ -0,0 +1,195 @@
+        ///// </summary>
+        ///// <param name="request"><see cref="System.Windows.Input.TraversalRequest"/> that contains requested navigation direction</param>
+        ///// <returns>Did handle tab</returns>
+        //protected override bool TabIntoCore(System.Windows.Input.TraversalRequest request)
+        //{
+        //    if (_xamlSource.HasFocus && !_onTakeFocusRequested)
+        //    {
+        //        return false; // If we have focus already, then we dont need to NavigateFocus
+        //    }
+
+        //    // Bug 17544829: Focus is wrong if the previous element is in a different FocusScope than the UnoXamlHost element.
+        //    var focusedElement = System.Windows.Input.FocusManager.GetFocusedElement(
+        //        System.Windows.Input.FocusManager.GetFocusScope(this)) as FrameworkElement;
+
+        //    var origin = BoundsRelativeTo(focusedElement, this);
+        //    var reason = MapDirectionToReason[request.FocusNavigationDirection];
+        //    if (_lastFocusRequest == Guid.Empty)
+        //    {
+        //        _lastFocusRequest = Guid.NewGuid();
+        //    }
+
+        //    var sourceFocusNavigationRequest = new WUX.Hosting.XamlSourceFocusNavigationRequest(reason, origin, _lastFocusRequest);
+        //    try
+        //    {
+        //        var result = _xamlSource.NavigateFocus(sourceFocusNavigationRequest);
+
+        //        // Returning true indicates that focus moved.  This will cause the HwndHost to
+        //        // move focus to the source’s hwnd (call SetFocus Win32 API)
+        //        return result.WasFocusMoved;
+        //    }
+        //    finally
+        //    {
+        //        _lastFocusRequest = Guid.Empty;
+        //    }
+        //}
+
+        /// <summary>
+        /// Transform bounds relative to FrameworkElement
+        /// </summary>
+        /// <param name="sibling1">base rectangle</param>
+        /// <param name="sibling2">second of pair to transform</param>
+        /// <returns>result of transformed rectangle</returns>
+        private static WF.Rect BoundsRelativeTo(FrameworkElement sibling1, System.Windows.Media.Visual sibling2)
+        {
+            WF.Rect origin = default(WF.Rect);
+
+            if (sibling1 != null)
+            {
+                // TransformToVisual can throw an exception if two elements don't have a common ancestor
+                try
+                {
+                    var transform = sibling1.TransformToVisual(sibling2);
+                    var systemWindowsRect = transform.TransformBounds(
+                        new Rect(0, 0, sibling1.ActualWidth, sibling1.ActualHeight));
+                    origin.X = systemWindowsRect.X;
+                    origin.Y = systemWindowsRect.Y;
+                    origin.Width = systemWindowsRect.Width;
+                    origin.Height = systemWindowsRect.Height;
+                }
+                catch (System.InvalidOperationException)
```

Comment:
🤖

Reply:
The suggestion is gone now, not sure what it was, but can refactor later :-D